src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatisticsModel.java

```diff
@@ -0,0 +1,116 @@
+import de.dennisguse.opentracks.util.UnitConversions;
+
+/**
+ * This view model is used to load intervals for a track.
+ * It uses a default interval but it can be set from outside to manage the interval length.
+ */
+public class IntervalStatisticsModel extends AndroidViewModel {
+
+    private MutableLiveData<IntervalStatistics> intervalStats;
+    private IntervalOption interval;
+
+    public IntervalStatisticsModel(@NonNull Application application) {
+        super(application);
+        interval = IntervalOption.OPTION_1;
+    }
+
+    public LiveData<IntervalStatistics> getIntervalStats(Track.Id trackId, IntervalOption interval) {
+        if (intervalStats == null || this.interval != interval) {
+            intervalStats = new MutableLiveData<>();
+            this.interval = interval;
+            loadIntervalStats(trackId);
+        }
+        return intervalStats;
+    }
+
+    public LiveData<IntervalStatistics> getIntervalStats(Track.Id trackId) {
+        return getIntervalStats(trackId, interval);
+    }
+
+    /**
+     * Call this method when you want to force the view model to re-load intervals.
+     */
+    public void invalidate() {
+        intervalStats = null;
+    }
+
+    private void loadIntervalStats(Track.Id trackId) {
+        new Thread(() -> {
+            Context context = getApplication().getApplicationContext();
+            ContentProviderUtils contentProviderUtils = new ContentProviderUtils(context);
+            List<TrackPoint> trackPointList = contentProviderUtils.getTrackPoints(trackId);
+
+            IntervalStatistics intervalStatistics = new IntervalStatistics();
+            float distanceInterval = PreferencesUtils.isMetricUnits(context) ? (float) (interval.getValue() * UnitConversions.KM_TO_M) : (float) (interval.getValue() * UnitConversions.MI_TO_M);
+            intervalStatistics.build(trackPointList, distanceInterval);
+
+            intervalStats.postValue(intervalStatistics);
+        }).start();
+    }
+
+    /**
+     * Intervals length this view model support.
+     */
+    public enum IntervalOption {
+        OPTION_1(1),
+        OPTION_2(2),
+        OPTION_3(3),
+        OPTION_4(4),
+        OPTION_5(5),
+        OPTION_10(10),
+        OPTION_20(20),
+        OPTION_50(50);
+
+        private int value;
+
+        IntervalOption(int value) {
+            this.value = value;
+        }
+
+        /**
+         * @param pos position of the interval option.
+         * @return    the interval option that is in the position pos.
+         */
+        public static IntervalOption getIntervalOption(int pos) {
+            if (values().length > pos) {
+                return values()[pos];
+            } else {
+                return OPTION_1;
+            }
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        /**
+         * @return a string array with all options.
+         */
+        public static String[] getAllValues() {
```

Comment:
This is just for to string conversion, right?
If yes, then I would move this functionality to the actual user of this function.

```
        int[] a = {1};
        Arrays.stream(a).mapToObj(String::valueOf).toArray(String[]::new);
```

Reply:
Is this okay?

Review this please:

```java
int[] intValues = Arrays.stream(IntervalStatisticsModel.IntervalOption.values()).mapToInt(i -> i.getValue()).toArray();
spinnerIntervals.setAdapter(new ArrayAdapter<>(getContext(), android.R.layout.simple_spinner_dropdown_item, Arrays.stream(intValues).mapToObj(String::valueOf).toArray(String[]::new)));
```

Reply:
Why not?
```
String[] values = Arrays.stream(IntervalStatisticsModel.IntervalOption.values()).mapToObj(String::valueOf).toArray(String[]::new))
spinnerIntervals.setAdapter(new ArrayAdapter<>(getContext(), android.R.layout.simple_spinner_dropdown_item, values);
```

Reply:
I didn't sure what you want was it.
Okay, then :+1:

Reply:
PS: Not tested ;)